### PR TITLE
Add Docker setup for jupyter notebooks

### PR DESCRIPTION
**Pull request by Baschdl**
Alternative to #49:
That's a docker image for jupyter notebooks building up on our other docker image

### CODE REVIEW

File: deployment/jupyter_notebook_config.py [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# Set ip to '*' to bind on all interfaces (ips) for the public server
+c.NotebookApp.ip = '*'
```
Comment (GittiHab) [inline deployment/jupyter_notebook_config.py]: That's what I probably forgot.

File: Dockerfile-Notebook [hunk 2]
```diff
@@ -0,0 +1,9 @@
+FROM server:latest
+EXPOSE 8888
+
+RUN apt-get update && apt-get install -y graphviz
+RUN pip3 install jupyter
+
+COPY . /32de-python/
+
+ENTRYPOINT ["jupyter", "notebook", "--config", "deployment/jupyter_notebook_config.py"]
```
Comment (GittiHab) [inline Dockerfile-Notebook]: I would at least add `tini` as this should make jupyter more stable – this is also recommended in the [Jupyter Docs](https://jupyter-notebook.readthedocs.io/en/stable/public_server.html#docker-cmd). You can also refer to the base [Jupyter Dockerfile](https://github.com/jupyter/docker-stacks/blob/master/base-notebook/Dockerfile)

Comment (GittiHab) [inline Dockerfile-Notebook]: I would at least add `tini` as this should make jupyter more stable – this is also recommended in the [Jupyter Docs](https://jupyter-notebook.readthedocs.io/en/stable/public_server.html#docker-cmd). You can also refer to the base [Jupyter Dockerfile](https://github.com/jupyter/docker-stacks/blob/master/base-notebook/Dockerfile)

Comment (Baschdl) [reply Dockerfile-Notebook]: Yes, I also read this. But they just mention it in combination with the `CMD` command. I didn't had problems yet, I would add it if the first error occurs.

**Comment by Baschdl** [comment]
Password is our standard password

File: Dockerfile-Notebook (hunk 2 above)
Comment (GittiHab) [reply Dockerfile-Notebook]: Can't you try putting it in and if it works, it works and if it doesn't you can leave it out for now? I wouldn't want to risk a kernel crash in the middle of a very interesting experiment.

**Comment by Baschdl** [comment]
Ok, I'll do it tomorrow

**Comment by Baschdl** [comment]
@GittiHab Do we also need this here? https://github.com/KDD-OpenSource/32de-python/blob/adf900b03effca6057849adc57b33dbca5b498fc/Dockerfile.jupyter#L10
What does it do?

**Comment by GittiHab** [comment]
Did you test it? Does it work? Then you could merge. Another thought would be that this might not be the right repo for this Docker Image and to move it to another (internal) repo.

**Comment by Baschdl** [comment]
Yes, it works. 
Do we also need this here? https://github.com/KDD-OpenSource/32de-python/blob/adf900b03effca6057849adc57b33dbca5b498fc/Dockerfile.jupyter#L10

**Comment by GittiHab** [comment]
> Do we also need this here? 

No, it should have fixed the module `gi` not being found by conda but it didn't help.

**Pull request merged**